packages/SwingSet/src/kernel/kernel.js

```diff
@@ -472,7 +472,16 @@ export default function buildKernel(
       kernelKeeper.incStat(statNameForNotify(p.state));
       const kd = harden(['notify', kpid, p]);
       const vd = vat.translators.kernelDeliveryToVatDelivery(kd);
-      await deliverAndLogToVat(vatID, kd, vd);
+      if (vd) {
+        await deliverAndLogToVat(vatID, kd, vd);
```

Comment:
yeah, as discussed in the meeting we should change `vatKeeper` to have a method like`deleteCListsForKpids(kpids)` or something, to allow the translation to take place as close as possible to the data it needs

Reply:
As I've been working through this change, I discovered that the notion of deleting the clist entries after translation but before delivery will not work given the current implementation of liveslots, because during the processing of the notify, the vat generates ephemeral subscribe requests that mention promises that are going away.  If the rule is "never speak of this again after this crank", then deleting the clist entries after the notify delivery (i.e., the way it works now) is fine, but if we want to have the vpids to immediately become unmentionable, we'll have to devise a different way to manage these ephemeral subscribe syscalls.  I'm inclined to want to do that, since these subscribes are kinda stupid, but the mechanism is not obvious.

Reply:
I think we worked this out, by having liveslots keep track of which promises were introduced during deserialization and which are being resolved by the crank, and only subscribing to the remainder. So the vat should never be subscribing to `vpid`s that are introduced-and-resolved.